src/layers/legacy/medVtkInria/vtkImageView/vtkImageViewCornerAnnotation.cxx

```diff
@@ -170,427 +170,183 @@ void vtkImageViewCornerAnnotation::TextReplace(vtkImageActor *ia,
         }
     }
 
-    // search for tokens, replace and then assign to TextMappers
-    for (i = 0; i < 4; i++)
+    int isKnownAnnotationStyle = -1;
+    if (view2d)
     {
-        if (this->CornerText[i] && strlen(this->CornerText[i]))
-        {
-            size_t len = strlen(this->CornerText[i]) + 1000;
-            text  = new char[len];
-            text2 = new char[len];
-            strncpy(text, this->CornerText[i], strlen(this->CornerText[i])+1);
-
-            // now do the replacements
+        isKnownAnnotationStyle = static_cast<int>(view2d->GetAnnotationStyle2D());
```

Comment:
this variable allow to know if the annotation style is known or not, in order to avoid unnecessary code.
There are two annotation style: vtkImageView2D::AnnotationStyle1 and vtkImageView2D::AnnotationStyle2 which can be seen in vtkImageView2D::SetAnnotationsFromOrientation().
The classic medInria style in the first one.

Reply:
ok